install/infra/single-cluster/azure/README.md

```diff
@@ -0,0 +1,225 @@
+# Terraform setup for AWS Single-cluster Gitpod reference architecture
+
+This directory has terraform configuration necessary to achieve a infrastructure
+corresponding to the Single-cluster reference architecture for Gitpod on Azure.
+
+This module will do the following steps:
+- Creates the infrastructure using our [`aks` terraform module](../../modules/aks/), which does:
+  - Setup an AKS managed cluster, along with external dependencies like database, storage and registry (if chosen)
+  - Sets up Azure Cloud DNS entries for the domain name (if chosen)
+- Provisioning the cluster:
+  - Set up cert-manager using our [`cert-manager` module](../../modules/tools/cert-manager/)
+  - Set up external-dns using our [`external-dns` module](../../modules/tools/external-dns/)
+  - Creates a cluster-issuer using our [`issuer` module](../../modules/tools/issuer/)
+
+> 💡 If you would like to create the infrastructure orchestrating the terraform modules by yourself, you can find all the modules we support [here](../../modules/).
+
+Since the entire setup requires more than one terraform target to be run due to
+dependencies (eg: helm provider depends on kubernetes cluster config, which is
+not available until the `aks` module finishes), this directory has a `Makefile`
+with targets binding together targeted terraform calls. This document will
+explain the execution of the terraform module in terms of these `make` targets.
+
+## Requirements
+
+* `terraform` >= `v1.1.0`
+* `kubectl`   >= `v1.20.0`
+* [`jq`](https://stedolan.github.io/jq/download/)
+
+## Setup Azure authentication and `azurerm` backend storage
+
+create IAM, set env vars, create backend bucket
+
+Before starting the installation process, you need:
+* An Azure account
+  - [Create one now by clicking here](https://azure.microsoft.com/en-gb/free/)
+- A user account with "Owner" IAM rights on the subscription
```

Comment:
**Question:** When I worked on the reference architecture guide I believe I used the `Contributor` level, can we double check the permission level?

Reply:
Good point! I have to be honest, I basically copied this bit from the [`gitpod-microsoft-aks-guide`](https://github.com/gitpod-io/gitpod-microsoft-aks-guide). I will check this

Reply:
So regarding this, I am not entirely sure if a fix is needed. Considering the client id and secret are relating to the tenant. Do you mind if we ship this as is for now? We will anyway have to iterate on this, when syncing with the reference architecture.

Reply:
Yeah, let's ship it 😄